services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsManagementService.java

```diff
@@ -43,21 +46,31 @@
 
     protected TenantInformationService tenantInformationService = new NoopTenantInformationService();
 
-    private HonoPasswordEncoder passwordEncoder;
-
     private final Vertx vertx;
+    private final HonoPasswordEncoder passwordEncoder;
+    private final int maxBcryptIteration;
+    private final Set<String> hashAlgorithmsWhitelist;
 
     /**
      * Creates a service for the given Vertx and password encoder instances.
      *
      * @param vertx The Vertx instance to use.
      * @param passwordEncoder The password encoder.
-     * @throws NullPointerException if any of the parameters is {@code null};
+     * @param maxBcryptIteration The maximum number of allowed bcrypt iterations.
+     * @param hashAlgorithmsWhitelist An optional collection of allowed password hashes.
+     * @throws NullPointerException if any of the required parameters is {@code null};
      */
-    @Autowired
-    public AbstractCredentialsManagementService(final Vertx vertx, final HonoPasswordEncoder passwordEncoder) {
+    public AbstractCredentialsManagementService(
+            final Vertx vertx,
+            final HonoPasswordEncoder passwordEncoder,
+            final int maxBcryptIteration,
+            final Set<String> hashAlgorithmsWhitelist) {
+
         this.vertx = Objects.requireNonNull(vertx);
         this.passwordEncoder = Objects.requireNonNull(passwordEncoder);
+        this.maxBcryptIteration = maxBcryptIteration;
```

Comment:
FMPOV we should also make sure that the iterations are within the supported value range, in our case [4, 31].
Having read throught the bcrypt documentation again, it seems like the parameter should better be called *maxBcryptLogRounds* as the number of actual iterations is computed as 2^logRounds ...

Reply:
I would say this is already covered by the configuration, which explicitly checks for that. Also would renaming mean, that we should do that consistently throughout the project. Not only here.

So I would say, we keep this PR focused on the current items. And create a new one later for renaming that parameter, which has an impact on documentation and configuration as well.

Reply:
fine with me